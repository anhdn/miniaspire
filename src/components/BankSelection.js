import React, { useState } from 'react'
import styled from 'styled-components'

const BankContainer = styled.div`
  padding: 5px 5px 5px 5p;
  margin-bottom: 20px;
`
const Title = styled.h2`
color: #292a2a;
font-size: 14px;
font-weight: 600;
margin-bottom: 9px;
padding-bottom: 9px;
`
const TextSelection = styled.div`
margin-bottom: 10px
`
const BankSelection = ({ onSelectedBank, onSelectedAmount, bankSelected }) => {
  const [bank, setBank] = useState('')
  const selectedBank = (bank) => {
    setBank(bank)
    onSelectedBank(bank)
  }
  const getAmount = (e) => { onSelectedAmount(e.target.value) }

  return (
    <BankContainer>
      <Title>Choose a bank</Title>
      <div className='cc-selector'>
        <input id='citibank' type='radio' name='bank' value='citibank' onClick={() => selectedBank(1)} />
        <label className='drinkcard-cc citibank' htmlFor='citibank' />
        <input id='hsbcbank' type='radio' name='bank' value='hsbc' onClick={() => selectedBank(2)} />
        <label className='drinkcard-cc hsbcbank' htmlFor='hsbcbank' />
        <input id='standardchartered' type='radio' name='bank' value='standardchartered' onClick={() => selectedBank(3)} />
        <label className='drinkcard-cc standardchartered' htmlFor='standardchartered' />
        {bank && (<TextSelection>Selected <strong>{bankSelected.name} - Interest rate {bankSelected.rate}%</strong></TextSelection>)}
        <Title >Loan Amount</Title>
        <div className='form-group input-group mb-2 mr-sm-2 mb-sm-0'>
          <input
            placeholder='10000 sgd'
            id='amount' type='text'
            name='amount'
            onChange={getAmount}
            className='form-control amount'
          />
          <div className='input-group-append'>
            <span className='input-group-text' id='basic-addon1'>sgd</span>
          </div>
        </div>
      </div>

    </BankContainer>

  )
}

export default BankSelection
