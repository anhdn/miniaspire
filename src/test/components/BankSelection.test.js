import React from 'react'
import { configure } from 'enzyme'
import BankSelection from '../../components/BankSelection'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

it('renders correctly without function handle change', () => {
  const tree = renderer
    .create(<BankSelection />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
