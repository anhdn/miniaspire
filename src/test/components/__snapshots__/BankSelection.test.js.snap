// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without function handle change 1`] = `
<div
  className="BankSelection__BankContainer-sc-1xcycdl-0 hHhsAf"
>
  <h2
    className="BankSelection__Title-sc-1xcycdl-1 jLJwjv"
  >
    Choose a bank
  </h2>
  <div
    className="cc-selector"
  >
    <input
      id="citibank"
      name="bank"
      onClick={[Function]}
      type="radio"
      value="citibank"
    />
    <label
      className="drinkcard-cc citibank"
      htmlFor="citibank"
    />
    <input
      id="hsbcbank"
      name="bank"
      onClick={[Function]}
      type="radio"
      value="hsbcbank"
    />
    <label
      className="drinkcard-cc hsbcbank"
      htmlFor="hsbcbank"
    />
    <input
      id="standardchartered"
      name="bank"
      onClick={[Function]}
      type="radio"
      value="standardchartered"
    />
    <label
      className="drinkcard-cc standardchartered"
      htmlFor="standardchartered"
    />
    
    <h2
      className="BankSelection__Title-sc-1xcycdl-1 jLJwjv"
    >
      Loan Amount
    </h2>
    <div
      className="form-group input-group mb-2 mr-sm-2 mb-sm-0"
    >
      <input
        className="form-control amount"
        id="amount"
        name="amount"
        onChange={[Function]}
        placeholder="10000 sgd"
        type="text"
      />
      <div
        className="input-group-append"
      >
        <span
          className="input-group-text"
          id="basic-addon1"
        >
          sgd
        </span>
      </div>
    </div>
  </div>
</div>
`;
